spec/requests/api/v2/providers/publish_spec.rb

```diff
@@ -36,16 +36,45 @@
 
     context 'unpublished provider with draft enrichment' do
       let(:enrichment) { build(:provider_enrichment, :initial_draft) }
-      let(:provider) {
+      let!(:provider) do
         create(
           :provider,
           organisations: [organisation],
-          enrichments: [enrichment]
+          enrichments: [enrichment],
+          courses: [create(:course), create(:course)]
```

Comment:
build vs create?

Reply:
build breaks it.

Reply:
odd enrichment uses build....

Reply:
is the course created assigns the current provider or a new provider?

Reply:
Oh hang on I thought you were talking about switching the provider to `build()`, switching the courses works fine.